beacon-chain/state/state-native/proofs.go

```diff
@@ -41,34 +57,44 @@ func (b *BeaconState) CurrentSyncCommitteeProof(ctx context.Context) ([][]byte,
 	if err := b.recomputeDirtyFields(ctx); err != nil {
 		return nil, err
 	}
-	return fieldtrie.ProofFromMerkleLayers(b.merkleLayers, currentSyncCommittee), nil
+	return fieldtrie.ProofFromMerkleLayers(b.merkleLayers, int(b.fieldIndexPosition[nativetypes.CurrentSyncCommittee])), nil
 }
 
 // NextSyncCommitteeProof from the state's Merkle trie representation.
 func (b *BeaconState) NextSyncCommitteeProof(ctx context.Context) ([][]byte, error) {
 	b.lock.Lock()
 	defer b.lock.Unlock()
+
+	if b.version == version.Phase0 {
+		return nil, fmt.Errorf("NextSyncCommitteeProof is not supported for %s", version.String(b.version))
```

Comment:
There are so many of these that I feel like having a helper 
```go
func ErrNotSupported( field string, vers int) error {
    return fmt.Errorf("%s is not supported for %s", field, version.String(vers))
}
```
Is useful cause on tests you can test against `require.ErrorIs(t, ErrNotSupported(field, b.version), err)`